sdk/include/opentelemetry/sdk/trace/recordable.h

```diff
@@ -41,6 +42,18 @@ class Recordable
                       opentelemetry::trace::SpanId span_id,
                       opentelemetry::trace::SpanId parent_span_id) noexcept = 0;
 
+  /**
+   * Sets the resource associated with a span.
+   *
+   * <p> The resource is guaranteed to have a lifetime longer than this recordable.  It is
+   * owned by the `TracerProvider` which (indirectly) owns the `Exporter` that generates a
+   * `Recordable`.
+   *
+   * @param resource pointer to the Resource for this span.
+   */
+  virtual void SetResourceRef(
+      const opentelemetry::sdk::resource::Resource *const resource) noexcept = 0;
```

Comment:
No strong preference, but can we pass const reference to resource here ?

Reply:
So I can do this, it's a bit "odd".  TL;DR: I didn't want to convert from a reference to a pointer as that causes me to cringe in the notion of "likely bug".  I found a way to not do that, but it's still a bit cringey, PTAL.